core/src/main/java/io/questdb/griffin/ExpressionParser.java

```diff
@@ -319,6 +319,46 @@ void parseExpr(GenericLexer lexer, ExpressionParserListener listener) throws Sql
                         }
                         break;
 
+                    case '#':
+                        if (GenericLexer.isGeoHashCharsConstant(tok)) { // e.g. #sp052w92p1p8
+                            thisBranch = BRANCH_CONSTANT;
+                            CharSequence geohashTok = GenericLexer.immutableOf(tok);
+                            // optional / bits '/dd', '/d'
+                            CharSequence slash = SqlUtil.fetchNext(lexer);
+                            if (slash == null || slash.charAt(0) != 47) { // '/'
+                                lexer.unparse();
+                                opStack.push(expressionNodePool.next().of(
```

Comment:
I would parse geohash here and push LONG constant here instead of creating immutable token